fees/euler/index.ts

```diff
@@ -0,0 +1,109 @@
+import { Adapter, FetchOptions, SimpleAdapter } from "../../adapters/types"
+import { CHAIN } from "../../helpers/chains"
+import * as sdk from "@defillama/sdk";
+
+const eVaultFactories = {
+    [CHAIN.ETHEREUM]: "0x29a56a1b8214D9Cf7c5561811750D5cBDb45CC8e",
+    [CHAIN.SONIC]: "0xF075cC8660B51D0b8a4474e3f47eDAC5fA034cFB",
+    [CHAIN.BASE]: "0x7F321498A801A191a93C840750ed637149dDf8D0"
+}
+
+const eulerFactoryABI = {
+    vaultLength: "function getProxyListLength() view returns (uint256)",
+    getProxyListSlice: "function getProxyListSlice(uint256 start, uint256 end) view returns (address[] list)",
+}
+
+const eulerVaultABI = {
+    asset: "function asset() view returns (address)",
+    interestAccumulator: "function interestAccumulator() view returns (uint256)",
+    accumulatedFeesAssets: "function accumulatedFeesAssets() view returns (uint256)",
+    totalBorrows: "function totalBorrows() view returns (uint256)",
+    convertToAssets: "function convertToAssets(uint256 shares) view returns (uint256)",
+    convertFees: "event ConvertFees(address indexed account, address indexed protocolReceiver, address indexed governorReceiver, uint256 protocolShares, uint256 governorShares)",
+    vaultStatus: "event VaultStatus(uint256 totalShares, uint256 totalBorrows, uint256 accumulatedFees, uint256 cash, uint256 interestAccumulator, uint256 interestRate, uint256 timestamp)"
+}
+
+const getVaults = async ({createBalances, api, fromApi, toApi, getLogs, chain}: FetchOptions, {
+    dailyFees,
+    dailyRevenue,
+}: {
+    dailyFees?: sdk.Balances,
+    dailyRevenue?: sdk.Balances,
+}) => {
+
+    if (!dailyFees) dailyFees = createBalances()
+    if (!dailyRevenue) dailyRevenue = createBalances()
+    const vaultLength = await api.call({target: eVaultFactories[chain], abi: eulerFactoryABI.vaultLength})
+    const vaults = await api.call({target: eVaultFactories[chain], abi: eulerFactoryABI.getProxyListSlice, params: [0, vaultLength]})
+    const underlyings = await api.multiCall({calls: vaults.map(vault=>({target: vault})), abi: eulerVaultABI.asset})
+    underlyings.forEach((underlying, index) => {
+        if (!underlying) underlyings[index] = '0x0000000000000000000000000000000000000000'
+    })    
+    
+    const vaultWithUnderlyings = vaults.map((vault, index) => ({vault, underlying: underlyings[index]}))
+    await Promise.all(vaultWithUnderlyings.map(async ({ vault, underlying }) => {
```

Comment:
we use .multiCall instead of Promise.all where we can as Promise.all leads to individual rpc calls, muticall batches everything into a single request